pkg/controller/migplan/validation.go

```diff
@@ -69,14 +72,15 @@ const (
 
 // Reasons
 const (
-	NotSet        = "NotSet"
-	NotFound      = "NotFound"
-	NotDistinct   = "NotDistinct"
-	LimitExceeded = "LimitExceeded"
-	NotDone       = "NotDone"
-	Done          = "Done"
-	Conflict      = "Conflict"
-	NotHealthy    = "NotHealthy"
+	NotSet                     = "NotSet"
+	NotFound                   = "NotFound"
+	NotDistinct                = "NotDistinct"
+	ProxySecretIsMisconfigured = "ProxySecretIsMisconfigured"
```

Comment:
The `reason` should describe the reason for the condition (or last state transition).

Perhaps:

`KeyError` when the secret is missing keys.
`Conflict` when multiple secrets found.

Reply:
Got it, thank you